docs/tests.md

```diff
@@ -1,33 +1,127 @@
-- **qualifiers**: the `qualifiers` corresponding to this `purl` as an object
-  of {key: value} qualifier pairs.
-- **subpath**: the `subpath` corresponding to this `purl`.
-- **is_invalid**: a boolean flag set to true if the test should report an
-  error
-
-To test `purl` parsing and building, a tool can use this test suite and for
-every listed test object, run these tests:
-
-- parsing the test canonical `purl` then re-building a `purl` from these
-  parsed components should return the test canonical `purl`
-
-- parsing the test `purl` should return the components parsed from the test
-  canonical `purl`
-
-- parsing the test `purl` then re-building a `purl` from these parsed
-  components should return the test canonical `purl`
-
-- building a `purl` from the test components should return the test canonical
-  `purl`
+The Package-URL (PURL) specification provides a JSON Schema and test
+files to support language-neutral testing of PURL implementations. The 
+objectives for the PURL schema and test files are to enable tools to conform
+to the PURL specification for tool functions such as:
+- build a canonical PURL string from a set of PURL component-level data
+- parse a canonical PURL string into a set of PURL components
+- validate an input PURL string and optionally provide warning or info 
+messages or correct errors in the input PURL string
+
+The current  JSON schema is available at: `purl-spec/schemas/purl-test.schema-0.1.json`.
+
+The test files are available at:
+- `purl-spec/tests/spec/`: This folder contains JSON test files that are not
+for a specific PURL type.
+  - `specification-test.json` - This file contains an array of test objects
+  that primarily cover testing the validity of individual PURL components or
+  separators between a pair of PURL components.
+  - component-`test.json`: These are test files for a specific PURL component.
+ See https://github.com/package-url/purl-spec/pull/738 which adds 
+ `tests/spec/qualifiers.json`.
+- `purl-spec/tests/types/`: This folder contains one JSON test file for each 
+registered PURL type. These tests should be focused on test cases that are
+specific to a PURL type, such as those for namespace or qualifiers.
+
+Two key properties in the PURL test JSON schema are:
+
+**Test groups**
+
+There are two PURL test groups:
+- **base**: Test group for base conformance tests. Base tests are pass/fail.
+- **advanced**: Test group for advanced tests. Advanced tests are more 
+permissive than base tests. They may provide severity messages or correct
+errors.
+
+*Discussion point:*
+- *Is there a better name than advanced for the second test
+group? perhaps Permissive? Or name the groups Strict and Permissive?]*
+
+**Test types**
+
+There are four PURL test types:
+- **build**: A test to build a canonical PURL output string from an input of 
+decoded PURL components. See also `/docs/how-build.md`.
+- **parse**: A test to parse decoded components from a canonical PURL 
```

Comment:
Parse tests historically do not need their inputs to be canonical.

Reply:
Agreed and not sure why they should be. 
Updated description of parse test type to: "A test to parse decoded components from a PURL input string."
from: "A test to parse decoded components from a canonical PURL input string."